### PR TITLE
Styling: integrating glamor based styling, removing fabric-core dependency

### PR DESCRIPTION
**Pull request by dzearing**
Addresses #983 

#### Description of changes

Problems we're aiming to solve:

1. Ensure all components work out of the box. No dependency on global css.
2. All css rules are versioned.
3. Open up a path for richer styling which has better control over rule specificity and RTL sensitivity.
4. Support a more robust theming approach.

This change does a couple of things:

1. It creates a dependency on the @uifabric/styling package. This package is still shuffling, but we want to take an early dependency on it and gradually migrate our CSS story.

2. It completely removes all global css references. There are zero references to ms-font-*, ms-u-animation-*, ms-icon-*. That means out of the box things will work correctly without needed stylesheets.

3. Redoes the button styles using glamor. The existing rules have remained intact; we just use glamor to register them. And, if you provide custom rules, we have an opportunity to mix those in, and ensure that we control specificity; the overrides always win.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -148,7 +157,7 @@ export class Breadcrumb extends BaseComponent<IBreadcrumbProps, IBreadcrumbState
   }
 
   @autobind
-  private _onOverflowClicked(ev: React.MouseEvent<HTMLElement>) {
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Breadcrumb/Breadcrumb.tsx]: Reason for dropping the type here?

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts [hunk 2]
```diff
@@ -12,6 +13,32 @@ export interface IButton {
   focus: () => void;
 }
 
+export interface IButtonStyles {
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: Some comments to help people realize which is the appropriate style to override for customization?

File: packages/office-ui-fabric-react/src/components/Button/IconButton/IconButton.styles.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  getTheme
+} from '../../../Styling';
+import {
+  getStyles as getBaseButtonStyles
+} from '../BaseButton.styles';
+
+const DEFAULT_BUTTON_HEIGHT = '32px';
+const DEFAULT_PADDING = '0 4px';
+
+export function getStyles(
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Button/IconButton/IconButton.styles.ts]: No memoize?

File: packages/utilities/src/rtl.ts [hunk 4]
```diff
@@ -40,15 +40,15 @@ export function getRTL(): boolean {
 /**
  * Sets the rtl state of the page (by adjusting the dir attribute of the html element.)
  */
-export function setRTL(isRTL: boolean) {
+export function setRTL(isRTL: boolean, avoidPersisting = false) {
```
Comment (christiango) [inline packages/utilities/src/rtl.ts]: :)

**Comment by micahgodbolt** [comment]
pulling this down for a look as well

File: packages/office-ui-fabric-react/src/components/Button/PrimaryButton/PrimaryButton.styles.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  ITheme,
+  getTheme,
+  mergeStyles,
+  mergeStyleSets
+} from '../../../Styling';
+import {
+  getStyles as getDefaultButtonStyles
+} from '../DefaultButton/DefaultButton.styles';
+
+export function getStyles(
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/PrimaryButton/PrimaryButton.styles.ts]: This feels like a bunch of boilerplate. theme, custom styles, palette...can we/should we abstract some of that away?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/PrimaryButton/PrimaryButton.styles.ts]: I'm not really sure. Styles are a function of the theme and of customStyles.

It is possible that we can pull the "global" theme, but it removes the ability to override the theme in a context.

It is possible to not take in customStyles, and to instead mix them in from within the component itself.

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/PrimaryButton/PrimaryButton.styles.ts]: not saying to change any of the functionality. custom themes are obviously important. Just curious if we can wrap it up and abstract what I'm sure will end up in just about every styles file. 

If you think this is about as abstract as you want to get right now, no worries. We can always tweak it later.

**Pull request merged**